### ISSUE TITLE
Storybook: Remove knobs dependencies from each stories

### ISSUE THREAD

**Issue by asudoh** [carbon-design-system/ibm-dotcom-library#2494]
### The problem

As of today, Storybook stories in our codebase runs `addon-knobs` code to run different configurations for each stories. However, such different configurations can be controlled only by Storybook knobs UI.

Refs: #2387 #2492

### The solution

Allow programatic control of such configurations, by moving knobs to Storybook parameters, like:

```javascript
const MyStory = ({ parameters }) => (
  <MyComponent myProp={paramaters?.props?.MyComponent?.myProp}
);

MyStory.story = {
  parameters: {
    knobs: {
      // We lazy-evaluate knobs here in a Storybook decorator that creates `parameters.props`
      MyComponent() {
        return {
          myProp: select('Prop (myProp)', ['foo', 'bar'], 'foo'),
        },
      },
    },
  },
};
```

Instead of:

```javascript
const MyStory = () => (
  <MyComponent myProp={select('Prop (myProp)', ['foo', 'bar'], 'foo')}
);
```

### Acceptance criteria

- [ ] Parameters that were defined by knobs can be controlled programmatically

**Comment by jeffchew** [comment]
@asudoh might be a good huddle topic, I'm trying to understand the benefit we'd get out of this?

**Comment by asudoh** [comment]
Good point @jeffchew - Let me queue this up. The benefit is that test code gets control over several parameters in story, instead of effectively being hard-coded with knob default values.
